keps/sig-node/2369-sizing-providers/README.md

```diff
@@ -0,0 +1,363 @@
+ 
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+  If a sizing provider plugin is enabled and it fails to fetch the sizing values, the kubelet will abort the startup and log the error in kubelet logs. Since the user has explicitly requested the sizing values to be fetched using the exec plugin, kubelet should not default to any other values. Defaulting to any other values and bringing up the kubelet may result in an unpredictable behavior, such as node lock ups, as the default sizing values may be not optimal. 
+
+  The SLO for successfully invoking sizing provider exec plugin should be based on the SLO for that plugin to fetch the sizing values.
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+
+  No.
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+This feature depends on the existence of a sizing provider plugin binary on the host and a configuration file for the plugin to be read by the kubelet.
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+
+  No
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+  It will add a new kubelet-level API. This API only contains a TypeMeta though and is
+not an object.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+
+  No, but a plugin implementation may choose to make API calls to a cloud provider.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+
+  No.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+
+  Use of the exec plugin may increase the startup time if the exec plugin invoked
+by kubelet takes a long time.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+
+  Possibly, it depends on how long it takes for the exec plugin to return during kubelet startup.
+
+## Implementation History
+
+
+## Drawbacks
+
+* exec plugins may be expensive to invoke by kubelet during the startup.
+* a poorly implemented exec plugin may halt the startup for the kubelet.
+
+## Alternatives
+
+### Add a built-in sizing provider in-tree.
+#### Pros
```

Comment:
Another alternative is what @derekwaynecarr [suggested](https://github.com/kubernetes/enhancements/pull/2370#issuecomment-776170612). We implemented a plugin based mechanism in scheduler using both intree and http. Using metadata server is more inline with cloudproviders like AWS do. We can perhaps add this as supported implementation when we graduate to beta?